Database

diogodinisf opened the pull request:
you can't make the auto merge. but comment please

diogodinisf committed:
first attempt to gets database work

diogodinisf committed:
dataaccess moved to a proper class

diogodinisf committed:
first atomic methods for permanent data access, non-structured

diogodinisf committed:
Merge branch 'develop' into database

# Conflicts:
#	conf
#	pom.xml
#	src/main/java/pt/dinis/main/App.java
#	src/main/java/pt/dinis/main/Dealer.java
#	src/main/resources/config.properties

diogodinisf committed:
first attempt to make a workerThread with permanent data access
moving files between paths
correction of some error of last merge to this branch

diogodinisf committed:
first attempt to do the server communication protocol

diogodinisf committed:
first attempt to do the server communication protocol
re-commit: some copy-paste stupid errors

diogodinisf committed:
Merge branch 'database' of github.com:dinis/raspberryGameSuite into database

# Conflicts:
#	src/main/java/pt/dinis/communication/ClientCommunicationProtocol.java

stupid idea of amend a commit that is already push to remote repo

diogodinisf committed:
A better approach of WorkerThread. Preparation of LoginWorkerThread to do the first transaction of project.
New DB management methods.

diogodinisf committed:
correction of some data access bugs and data base creation

TijoloAzul commented on databasescripts/03-02-2017_db_creation:
```diff
@@ -0,0 +1,15 @@
+CREATE DATABASE game_suite
+  WITH OWNER = diogo
```
Please, do not use your name.
Sugestions: postgres, raspberry, admin...

TijoloAzul commented on databasescripts/03-02-2017_db_creation:
```diff
@@ -0,0 +1,15 @@
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) UNIQUE NOT NULL, 
+    password VARCHAR(255) NOT NULL,
+    date timestamp without time zone NOT NULL DEFAULT now()
```
creation_date, date is vague and could create conflicts

diogodinisf replied:
feito

TijoloAzul commented on src/main/java/pt/dinis/client/login/LoginClientCommunicationProtocol.java:
```diff
@@ -1,9 +1,7 @@
 import pt.dinis.main.Display;
-
-import java.util.*;
+import java.util.*; //?
```
remove "//?"

TijoloAzul commented on src/main/java/pt/dinis/communication/ClientCommunicationThread.java:
```diff
@@ -44,8 +43,9 @@ public void run() {
-                    WorkerThread temporaryThread = new WorkerThread(message, id);
-                    temporaryThread.run();
+                    if (!ClientCommunicationProtocol.protocol(message)) {
+                        logger.info("Can't process the message.");
```
put message at logger

TijoloAzul commented on src/main/java/pt/dinis/dataaccess/User.java:
```diff
@@ -0,0 +1,95 @@
+ */
+public class User {
+    //TODO
+    // getDate
```
why getDate?

diogodinisf replied:
se não vou usar a data para nada porque a quero guardar?
e se a quero guardar porque poderei precisar dela também acho interessante terjá um método preparado para a ir buscar

TijoloAzul replied:
podes guardar a data a pensar que só vais precisar dela acedendo à BD
mais, a minha perspectiva é diferente entre código - BD
No código tenho uma perspectiva de **se não se vê utilidade não deve lá estar**
Na BD, **se um dia mais tarde poderá ser importante e não se consegue reconstituir fica**

TijoloAzul commented on src/main/java/pt/dinis/dataaccess/User.java:
```diff
@@ -0,0 +1,95 @@
+            }
+        }
+
+        throw new NotFoundException();
```
we do not need so many methods... but it's OK if we want to play around

TijoloAzul commented on src/main/java/pt/dinis/exceptions/NotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * Created by diogo on 04-02-2017.
+ */
+public class NotFoundException extends Exception {
```
cool

TijoloAzul commented on src/main/java/pt/dinis/temporary/WorkerThread.java:
```diff
@@ -1,40 +1,58 @@
+
+            connection.getConnection().commit();
+            logger.info("New database transaction committed.");
+        } catch (SQLException e) {
```
any exception, you never know!

TijoloAzul commented on src/main/java/pt/dinis/temporary/WorkerThread.java:
```diff
@@ -1,40 +1,58 @@
+                connection.closeConnection();
+            } catch (SQLException e) {
+                logger.warn("Can't close the database connection.");
+            }
```
nice

TijoloAzul commented on src/main/java/pt/dinis/temporary/WorkerThread.java:
```diff
@@ -1,40 +1,58 @@
+        } catch (SQLException e) {
+            logger.warn("Can't open the database connection.");
+            return;
+        }
```
shouldn't you try to close your connection? just for precaution

TijoloAzul commented on src/main/java/pt/dinis/temporary/WorkerThreadDeprecated.java:
```diff
@@ -0,0 +1,40 @@
+package pt.dinis.temporary;
```
git is your friend: remove this

diogodinisf replied:
ok

TijoloAzul commented on src/main/resources/application.conf:
```diff
@@ -1,7 +1,10 @@
+user.properties.file.name = user.sample.conf
+db.path = game_suite
+db.user = diogo
+db.pass = diogo
```
not your name please

TijoloAzul commented on src/main/resources/application.conf:
```diff
@@ -1,7 +1,10 @@
 server.port = 1500
 server.hostname = localhost
-user.properties.file.name = user.conf
+user.properties.file.name = user.sample.conf
```
you shouldn't use the user.sample.conf

diogodinisf replied:
sim, faz sentido

TijoloAzul commented:
it's looking nice

Commit:
some request and interesting changesto do the pull request

Commit:
Merge branch 'database' of github.com:dinis/raspberryGameSuite into database

Commit:
correction of correction to pull request

diogodinisf committed:
new changes on sql things to pull request

diogodinisf committed:
remove some desnecessary comments

TijoloAzul committed:
Merge branch 'develop' into database

TijoloAzul commented on databasescripts/17-02-2017_rename_date_field:
```diff
@@ -0,0 +1 @@
+ALTER TABLE users RENAME COLUMN date TO creation_date; 
```
ok... ficamos com os underscores??
já agora, username é uma palavra só?

TijoloAzul commented on src/main/java/pt/dinis/dataaccess/DBConnection.java:
```diff
@@ -0,0 +1,42 @@
+package pt.dinis.dataaccess;
```
estou a usar a notação data.access

TijoloAzul commented:
No geral, 
- [ ] ~~O servidor agora não faz nada em resposta aos pedidos do cliente~~ [Resolver em #9]
- [x] Perceber o que acontece quando há acessos à BD em simultâneo
- [x] Acrescentar as configurações no gradle  [@TijoloAzul]
- [x] Haver login funcional [@DiogoDinisF]

TijoloAzul committed:
reorganizing sql creation scripts
add gradle reference

TijoloAzul committed:
WorkerThread was closing connection right after opening it
DBConnection was been passed everywhere, now we pass Connection

diogodinisf committed:
change data access folder name

diogodinisf committed:
change data access folder name - correctly

diogodinisf committed:
samething, changing folders

TijoloAzul commented on src/main/java/pt/dinis/communication/ClientCommunicationThread.java:
```diff
@@ -44,8 +43,9 @@ public void run() {
-                    WorkerThread temporaryThread = new WorkerThread(message, id);
-                    temporaryThread.run();
+                    if (!ClientCommunicationProtocol.protocol(message)) {
+                        logger.info("Can't process the message" + message);
```
"Cannot process message " + message

TijoloAzul commented on src/main/java/pt/dinis/main/App.java:
```diff
@@ -12,6 +16,14 @@ public static void main( String[] args ) throws IOException {
+        try {
+            DBConnection.prepareDBAccess();
+        } catch (ClassNotFoundException e) {
+            logger.fatal("Can't find postgreSQL driver.");
```
logger.fatal("...", e)

TijoloAzul committed:
some log message are slightly different
adding stacktrace to some logs
DBConnection and User are in folder pt.dinis.data.access

And it resolves #10
